circuit_defs/bigint_with_control/verifier/src/lib.rs

```diff
@@ -256,29 +326,61 @@ pub unsafe fn verify_with_configuration<I: NonDeterminismSource, V: LeafInclusio
     let mut fri_folding_challenges: [Mersenne31Quartic; NUM_FRI_STEPS] =
         [MaybeUninit::uninit().assume_init(); NUM_FRI_STEPS];
 
-    for (caps, challenge) in skeleton
+    for (((caps, challenge), pow_challenge), pow_bits) in skeleton
```

Comment:
As we have too many zip-s here, and we know that iterator lengths should be same size, it's better to change it into index-based iteration